### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vladyslav-Banul**

### CODE REVIEW

**Review by Denis-Demyanok — APPROVED** [review]

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,18 @@
...
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const appropriateMan = people.filter((person) => {
+    return person.sex === 'm'
+      && (century === undefined
+        ? true
+        : Math.ceil(person.died / 100) === century);
+  });
+
+  const res = appropriateMan.reduce((sum, currentValue) => {
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:25]: Let's create more meaningful variable name instead of 'res'

File: src/averageAges.js [hunk 2]
```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const appropriateMan = people.filter((person) => {
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:18]: 'Man' - singular form, 'men' - plural form
```suggestion
  const appropriateMen = people.filter((person) => {
```

File: src/averageAges.js [hunk 3]
```diff
@@ -36,8 +43,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  const appropriateWoman = people.filter((person) => {
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:47]: 'Woman' - singular form, 'women' - plural form
```suggestion
  const appropriateWomen = people.filter((person) => {
```

File: src/averageAges.js [hunk 4]
```diff
@@ -36,8 +43,19 @@ function calculateMenAverageAge(people, century) {
...
+    return person.sex === 'f'
+      && (withChildren
+        ? people.some((potentialChild) => potentialChild.mother === person.name)
+        : people);
+  });
+
+  const res = appropriateWoman.reduce((sum, currentValue) => {
```
Comment (Anna-Tkachenko) [inline src/averageAges.js:54]: Let's create more meaningful variable names

**Review by Anna-Tkachenko — APPROVED** [review]
Good job!👍

**Pull request closed without merging**